backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -264,30 +276,20 @@ public Response deleteEndpoint(@Context SecurityContext sec, @PathParam("id") UU
         EndpointType endpointType = endpointRepository.getEndpointTypeById(principal.getAccount(), id);
         checkSystemEndpoint(endpointType);
 
-        if (obEnabled) {
-            Endpoint e = endpointRepository.getEndpoint(principal.getAccount(), id);
-            if (e != null) {
-                EndpointProperties properties = e.getProperties();
-                if (properties instanceof CamelProperties) {
-                    CamelProperties cp = (CamelProperties) properties;
-                    // Special case wrt OpenBridge
-                    if (e.getSubType().equals("slack")) {
-                        String processorId = cp.getExtras().get(OB_PROCESSOR_ID);
-                        if (processorId != null) { // Should not be null under normal operations.
-                            try {
-                                bridgeApiService.deleteProcessor(bridge.getId(), processorId, bridgeAuth.getToken());
-                            } catch (Exception ex) {
-                                Log.warn("Removal of OB processor failed:" + ex.getMessage());
-                                // Nothing more we can do
-                            }
-                        } else {
-                            Log.warn("ProcessorId was null for endpoint " + id.toString());
-                        }
-                    }
-                }
+        Endpoint ep = endpointRepository.getEndpoint(principal.getAccount(), id);
+        if (isForOpenBridge(ep)) {
+
+            try {
+                UnaryOperator<Processor> execFun = in -> {
+                    bridgeApiService.deleteProcessor(bridge.getId(), in.getId(), bridgeAuth.getToken());
+                    return null;
+                };
+                UnaryOperator<Processor> updateFun = in -> { ep.setStatus(EndpointStatus.DELETING); return in; };
+                executeObAction(ep, updateFun, execFun);
```

Comment:
The code would be easier to understand/maintain if it looked like this:
```suggestion
                ep.setStatus(EndpointStatus.DELETING);
                executeObAction(ep, null, execFun); // TODO Handle null updateFun in executeObAction
```